Extensions/BBText/bbtextruntimeobject.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * An object displaying a BBCode formated text on the screen.
+ * @memberof gdjs
+ * @class BBTextRuntimeObject
+ * @extends RuntimeObject
+ */
+gdjs.BBTextRuntimeObject = function(runtimeScene, objectData) {
+  gdjs.RuntimeObject.call(this, runtimeScene, objectData);
+
+  console.log('OBJ:', this, objectData);
+  /** @type number */
+  this._opacity = objectData.content.opacity;
+  /** @type boolean */
+  this._visible = objectData.content.visible;
+  /** @type string */
+  this._text = objectData.content.text;
+  /** @type string */
+  this._color = objectData.content.color;
+  /** @type string */
+  this._family = objectData.content.family;
+  /** @type number */
+  this._size = objectData.content.size;
+  /** @type number */
+  this._wrappingWidth = objectData.content.width;
+  /** @type string */
+  this._align = objectData.content.align;
+
+  if (this._renderer)
+    gdjs.BBTextRuntimeObjectRenderer.call(this._renderer, this, runtimeScene);
+  else
+    this._renderer = new gdjs.BBTextRuntimeObjectRenderer(this, runtimeScene);
+
+  // *ALWAYS* call `this.onCreated()` at the very end of your object constructor.
+  this.onCreated();
+};
+
+gdjs.BBTextRuntimeObject.prototype = Object.create(
+  gdjs.RuntimeObject.prototype
+);
+gdjs.BBTextRuntimeObject.thisIsARuntimeObjectConstructor = 'BBText::BBText';
+
+gdjs.BBTextRuntimeObject.prototype.getRendererObject = function() {
+  return this._renderer.getRendererObject();
+};
+
+/**
+ * Initialize the extra parameters that could be set for an instance.
+ * @private
+ */
+gdjs.BBTextRuntimeObject.prototype.extraInitializationFromInitialInstance = function(
+  initialInstanceData
+) {
+  if (initialInstanceData.customSize) {
+    this.setWrappingWidth(initialInstanceData.width);
+  }
+};
+
+gdjs.BBTextRuntimeObject.prototype.onDestroyFromScene = function(runtimeScene) {
+  gdjs.RuntimeObject.prototype.onDestroyFromScene.call(this, runtimeScene);
+
+  this._renderer.onDestroy();
+};
+
+gdjs.BBTextRuntimeObject.prototype.update = function(runtimeScene) {
+  this._renderer.ensureUpToDate();
+};
+
+/**
+ * Set BBText base style properties
+ */
+gdjs.BBTextRuntimeObject.prototype.setBBText = function(text) {
+  this._renderer.setBBText(text);
```

Comment:
The renderer responsibility should be to only be "displaying" the object on the screen. As such, it should have little "state" as possible. The object should instead be the "source of truth", meaning that the object should be self sufficient and have (almost) everything that the renderer needs (I say almost because the renderer can contain Pixi specific stuff or memorize some things if needed).

All of this to say: you're calling `setBBText` on the renderer - which looks ok - but you're not updated `this._text`. If I'm not mistaken, the BBTextRuntimeObject should always contain the text that is displayed on screen :) 
The renderer will need it too, but it could almost read this by itself (that's the reason I chose names like "updatePosition", "updateOpacity" for renderers, instead of setX, setY).
You can see the renderer as something that is "reading" the object and displaying it on screen (using Pixi, but could be anything else).

Anyway, double check that your object is properly updating it properties when you're updating something :)

Reply:
I refactored it now so the renderer only updates, doesnt do any setting